DKMS build errors on Fedora 35 5.17.4-200

I started seeing the following build error for the kernel module on Fedora 35 5.17.4-200, which has been released a week ago as an update to 5.16.20:
```
DKMS make.log for jool-4.1.8 for kernel 5.17.4-200.fc35.x86_64 (x86_64)
Thu Apr 28 01:36:33 AM CEST 2022
make: Entering directory '/usr/src/kernels/5.17.4-200.fc35.x86_64'
CC [M]  /var/lib/dkms/jool/4.1.8/build/src/mod/common/rfc7915/4to6.o
/var/lib/dkms/jool/4.1.8/build/src/mod/common/rfc7915/4to6.c: In function ‘allocate_fast’:
/var/lib/dkms/jool/4.1.8/build/src/mod/common/rfc7915/4to6.c:407:9: error: implicit declaration of function ‘nf_reset’ [-Werror=implicit-function-declaration]
407 |         nf_reset(out);
|         ^~~~~~~~
cc1: some warnings being treated as errors
make[1]: *** [scripts/Makefile.build:288: /var/lib/dkms/jool/4.1.8/build/src/mod/common/rfc7915/4to6.o] Error 1
make: *** [Makefile:1841: /var/lib/dkms/jool/4.1.8/build/src/mod/common] Error 2
make: Leaving directory '/usr/src/kernels/5.17.4-200.fc35.x86_64'
```

My guess is that with the upgrade to 5.17 for F35 the changes got backported that changed `nf_reset` to `nf_reset_ct` and so these two conditionals need to be adjusted:
https://github.com/NICMx/Jool/blob/6822bdee4ec63467e82d723a3381b3116c1853d9/src/mod/common/rfc7915/4to6.c#L404-L408
https://github.com/NICMx/Jool/blob/6822bdee4ec63467e82d723a3381b3116c1853d9/src/mod/common/rfc7915/6to4.c#L470-L474

After writing the above I manually removed the preprocessor makro and ran it again, this time it failed at:
```
/var/lib/dkms/jool/4.1.8/build/src/mod/common/skbuff.c: In function ‘print_shinfo_fields’:
/var/lib/dkms/jool/4.1.8/build/src/mod/common/skbuff.c:472:50: error: ‘skb_frag_t’ {aka ‘struct bio_vec’} has no member named ‘page_offset’; did you mean ‘bv_offset’?
472 |                                 shinfo->frags[f].page_offset,
|                                                  ^~~~~~~~~~~
./include/linux/printk.h:418:33: note: in definition of macro ‘printk_index_wrap’
418 |                 _p_func(_fmt, ##__VA_ARGS__);                           \
|                                 ^~~~~~~~~~~
./include/linux/printk.h:531:9: note: in expansion of macro ‘printk’
531 |         printk(KERN_CONT fmt, ##__VA_ARGS__)
|         ^~~~~~
/var/lib/dkms/jool/4.1.8/build/src/mod/common/skbuff.c:38:17: note: in expansion of macro ‘pr_cont’
38 |                 pr_cont(text "\n", ##__VA_ARGS__); \
|                 ^~~~~~~
/var/lib/dkms/jool/4.1.8/build/src/mod/common/skbuff.c:468:17: note: in expansion of macro ‘print’
468 |                 print(tabs, "%u page_offset:%u size:%u", f,
|                 ^~~~~
make[1]: *** [scripts/Makefile.build:288: /var/lib/dkms/jool/4.1.8/build/src/mod/common/skbuff.o] Error 1
make: *** [Makefile:1841: /var/lib/dkms/jool/4.1.8/build/src/mod/common] Error 2
```
https://github.com/NICMx/Jool/blob/5dc6ae4fbc1620d3c6cefb8b98f054bb81c47a2a/src/mod/common/skbuff.c#L469-L473

and
```
/var/lib/dkms/jool/4.1.8/build/src/mod/common/error_pool.c:5:10: fatal error: stdarg.h: No such file or directory
5 | #include <stdarg.h>
|          ^~~~~~~~~~
compilation terminated.
make[1]: *** [scripts/Makefile.build:288: /var/lib/dkms/jool/4.1.8/build/src/mod/common/error_pool.o] Error 1
make: *** [Makefile:1841: /var/lib/dkms/jool/4.1.8/build/src/mod/common] Error 2
```
https://github.com/NICMx/Jool/blob/6822bdee4ec63467e82d723a3381b3116c1853d9/src/mod/common/error_pool.c#L4-L6

and
```
/var/lib/dkms/jool/4.1.8/build/src/mod/common/db/pool4/rfc6056.c: In function ‘rfc6056_f’:
/var/lib/dkms/jool/4.1.8/build/src/mod/common/db/pool4/rfc6056.c:170:13: error: ‘struct shash_desc’ has no member named ‘flags’
170 |         desc->flags = 0;
|             ^~
make[1]: *** [scripts/Makefile.build:288: /var/lib/dkms/jool/4.1.8/build/src/mod/common/db/pool4/rfc6056.o] Error 1
make: *** [Makefile:1841: /var/lib/dkms/jool/4.1.8/build/src/mod/common] Error 2
```
https://github.com/NICMx/Jool/blob/6822bdee4ec63467e82d723a3381b3116c1853d9/src/mod/common/db/pool4/rfc6056.c#L168-L171



Quick links in case they help:
- https://src.fedoraproject.org/rpms/kernel/commits/f35
- https://bodhi.fedoraproject.org/updates/?packages=kernel
- https://gitlab.com/cki-project/kernel-ark/-/tree/fedora-5.17
- https://gitlab.com/cki-project/kernel-ark/-/blob/fedora-5.17/include/linux/skbuff.h#L4432

--- a/src/mod/common/db/pool4/rfc6056.c
+++ b/src/mod/common/db/pool4/rfc6056.c
@@ -166,7 +166,7 @@ int rfc6056_f(struct xlation *state, unsigned int *result)
 
 	desc->tfm = shash;
 /* Linux commit: 877b5691f27a1aec0d9b53095a323e45c30069e2 */
-#if LINUX_VERSION_LOWER_THAN(5, 2, 0, 9999, 0)
+#if LINUX_VERSION_LOWER_THAN(5, 2, 0, 9, 99)
 	desc->flags = 0;
 #endif
 

--- a/src/mod/common/error_pool.c
+++ b/src/mod/common/error_pool.c
@@ -1,7 +1,7 @@
 #include "mod/common/error_pool.h"
 
 #include "mod/common/linux_version.h"
-#if LINUX_VERSION_LOWER_THAN(5, 15, 0, 9999, 0)
+#if LINUX_VERSION_LOWER_THAN(5, 15, 0, 9, 99)
 #include <stdarg.h>
 #endif
 

--- a/src/mod/common/rfc7915/4to6.c
+++ b/src/mod/common/rfc7915/4to6.c
@@ -401,7 +401,7 @@ static verdict allocate_fast(struct xlation *state, bool ignore_df,
 	}
 
 	/* https://github.com/NICMx/Jool/issues/289 */
-#if LINUX_VERSION_AT_LEAST(5, 4, 0, 9999, 0)
+#if LINUX_VERSION_AT_LEAST(5, 4, 0, 9, 99)
 	nf_reset_ct(out);
 #else
 	nf_reset(out);

--- a/src/mod/common/rfc7915/6to4.c
+++ b/src/mod/common/rfc7915/6to4.c
@@ -467,7 +467,7 @@ static verdict ttp64_alloc_skb(struct xlation *state)
 	}
 
 	/* https://github.com/NICMx/Jool/issues/289 */
-#if LINUX_VERSION_AT_LEAST(5, 4, 0, 9999, 0)
+#if LINUX_VERSION_AT_LEAST(5, 4, 0, 9, 99)
 	nf_reset_ct(out);
 #else
 	nf_reset(out);

--- a/src/mod/common/skbuff.c
+++ b/src/mod/common/skbuff.c
@@ -466,7 +466,7 @@ static void print_shinfo_fields(struct sk_buff *skb, unsigned int tabs)
 	tabs++;
 	for (f = 0; f < shinfo->nr_frags; f++) {
 		print(tabs, "%u page_offset:%u size:%u", f,
-#if LINUX_VERSION_AT_LEAST(5, 4, 0, 9999, 0)
+#if LINUX_VERSION_AT_LEAST(5, 4, 0, 9, 99)
 				skb_frag_off(&shinfo->frags[f]),
 #else
 				shinfo->frags[f].page_offset,

